clippy.toml

```diff
@@ -6,7 +6,7 @@ type-complexity-threshold = 500
 
 # Performance tuning
 avoid-breaking-exported-api = false
-msrv = "1.82.0"
+msrv = "1.87.0"
```

Comment:
Can we use a lower msrv? One reasonable approach to identify a good basis would be:
- select the highest msrv of all dependencies
- add an additional CI check to confirm that all CI requirements are met when the project is running with that rust version. 

This test is probably best-suited to a child PR.